index.bs

```diff
@@ -514,4 +514,47 @@ See [[FS#sandboxed-filesystem]].
 
 </aside>
 
+<h3 id="storage-bucket-clear-site-data">Clear Site Data integration</h3>
+
+Issue: Update [[clear-site-data#header]].
+
+: "<dfn grammar>`storage:bucket-name`</dfn>"
+
+  ::  If the type string starts with "`storage:`" then the remaining characters after the
+      `:` will be taken to refer to a specific [=storage bucket=] in |origin|.
+
+Issue: Update [[clear-site-data#parsing]] with these partial algorithms.
+
+<div algorithm>
+
+To <dfn>parse a Clear-Site-Data header with buckets</dfn>, execute the following steps:
+
+1. For each |type| in |header|, execute the following steps:
+
+    1. If |type| does not [=string/start with=] `"storage:"`, abort these steps.
+
+    1. Let |bucket name| be the [=code-unit-substring-by-positions|code unit substring=] from 8 to to |end| of |type|.
+
+    1. If the result of [=validate a bucket name=] with |bucket name| is failure, then abort these steps.
+
+    1. Append a [=tuple=] consisting of (`"storage-bucket"`, |bucket name|) to |types|.
```

Comment:
Maybe `Let |types| be an empty list` at the beginning to define this `types`?

Reply:
These steps are meant to be inserted into the existing algorithm that defines |types|. I tried to make that more clear in the `Issue` notes. WDYT?

Reply:
Ah I see, thanks for explaining. Updates to `Issue` lgtm!